extensions/standard-processors/processors/ListenSyslog.cpp

```diff
@@ -17,318 +14,283 @@
-      break;
-    if (retval == 0)
-      continue;
-    if (FD_ISSET(_serverSocket, &fds)) {
-      // server socket, either we have UDP datagram or TCP connection request
-      if (_protocol == "TCP") {
-        socklen_t clilen;
-        struct sockaddr_in cli_addr;
-        clilen = sizeof(cli_addr);
-        int newsockfd = accept(_serverSocket, reinterpret_cast<struct sockaddr *>(&cli_addr), &clilen);
-        if (newsockfd > 0) {
-          if (_clientSockets.size() < (uint64_t) _maxConnections) {
-            _clientSockets.push_back(newsockfd);
-            logger_->log_info("ListenSysLog new client socket %d connection", newsockfd);
-            continue;
-          } else {
-            close(newsockfd);
-          }
-        }
-      } else {
-        socklen_t clilen;
-        struct sockaddr_in cli_addr;
-        clilen = sizeof(cli_addr);
-        int recvlen = recvfrom(_serverSocket, _buffer, sizeof(_buffer), 0, (struct sockaddr *) &cli_addr, &clilen);
-        if (recvlen > 0 && (uint64_t) (recvlen + getEventQueueByteSize()) <= static_cast<uint64_t>(_recvBufSize)) {
-          uint8_t *payload = new uint8_t[recvlen];
-          memcpy(payload, _buffer, recvlen);
-          putEvent(payload, recvlen);
-        }
-      }
-    }
-    it = _clientSockets.begin();
-    while (it != _clientSockets.end()) {
-      int clientSocket = *it;
-      if (FD_ISSET(clientSocket, &fds)) {
-        int recvlen = readline(clientSocket, _buffer, sizeof(_buffer));
-        if (recvlen <= 0) {
-          close(clientSocket);
-          logger_->log_debug("ListenSysLog client socket %d close", clientSocket);
-          it = _clientSockets.erase(it);
-        } else {
-          if ((uint64_t) (recvlen + getEventQueueByteSize()) <= static_cast<uint64_t>(_recvBufSize)) {
-            uint8_t *payload = new uint8_t[recvlen];
-            memcpy(payload, _buffer, recvlen);
-            putEvent(payload, recvlen);
-          }
-          ++it;
-        }
-      }
-    }
+  if (!context->getProperty(MaxBatchSize.getName(), max_batch_size_) || max_batch_size_ <= 0) {
+    max_batch_size_ = MaxBatchSize.getDefaultValue();
+    logger_->log_debug("Missing or invalid attribute %s", MaxBatchSize.getName());
   }
-  return;
-}
 
-int ListenSyslog::readline(int fd, char *bufptr, size_t len) {
-  char *bufx = bufptr;
-  static char *bp;
-  static int cnt = 0;
-  static char b[2048];
-  char c;
-
-  while (--len > 0) {
-    if (--cnt <= 0) {
-      cnt = recv(fd, b, sizeof(b), 0);
-      if (cnt < 0) {
-        if (errno == EINTR) {
-          len++; /* the while will decrement */
-          continue;
-        }
-        return -1;
-      }
-      if (cnt == 0)
-        return 0;
-      bp = b;
-    }
-    c = *bp++;
-    *bufptr++ = c;
-    if (c == '\n') {
-      *bufptr = '\n';
-      return bufptr - bufx + 1;
-    }
+  if (!context->getProperty(ParseMessages.getName(), parse_messages_)) {
+    parse_messages_ = ParseMessages.getDefaultValue();
```

Comment:
isn't the default value returned when a property is not set?

Reply:
Didnt know that, but you are right.
Changed it in https://github.com/apache/nifi-minifi-cpp/pull/1294/commits/027fea1d65bd423940edd81d598cef68946e2202#diff-a22ae2c37869a184bd936ac719f98ae8bbdb1b8d2ad5b3e6fc48ea9207fe530bR88-R93